File: go/problems/greatest_number_of_candies.go
package problems

// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/
func KidsWithCandies(candies []int, extraCandies int) []bool {
	threshold := findMax(candies)
	candiesWithExtraOnes := make([]bool, len(candies))
	for idx, candy := range candies {
		finalSum := candy + extraCandies
		if finalSum >= threshold {
			candiesWithExtraOnes[idx] = true
		} else {
			candiesWithExtraOnes[idx] = false
		}
	}
	return candiesWithExtraOnes
}

func findMax(candies []int) int {
	max := candies[0]
	for _, candy := range candies {
		if candy > max {
			max = candy
		}
	}
	return max
}

File: rust/src/problems/greatest_number_of_candies.rs
// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/
pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
  let threshold: i32 = *candies.iter().max().unwrap();
  let candies_with_extra_ones: Vec<bool> = candies
    .iter()
    .map(|candy| {
      let final_sum: i32 = candy + extra_candies;
      return if final_sum >= threshold { true } else { false };
    })
    .collect();
  return candies_with_extra_ones;
}
